packages/rest/__tests__/REST.test.ts

```diff
@@ -1,209 +1,245 @@
 import { DiscordSnowflake } from '@sapphire/snowflake';
 import { Routes, Snowflake } from 'discord-api-types/v10';
-import nock from 'nock';
-import { Response } from 'node-fetch';
-import { REST, DefaultRestOptions, APIRequest } from '../src';
+import { File, FormData, MockAgent, setGlobalDispatcher } from 'undici';
+import type { Interceptable, MockInterceptor } from 'undici/types/mock-interceptor';
+import { genPath } from './util';
+import { REST } from '../src';
 
 const newSnowflake: Snowflake = DiscordSnowflake.generate().toString();
 
 const api = new REST().setToken('A-Very-Fake-Token');
 
-nock(`${DefaultRestOptions.api}/v${DefaultRestOptions.version}`)
-	.get('/simpleGet')
-	.reply(200, { test: true })
-	.delete('/simpleDelete')
-	.reply(200, { test: true })
-	.patch('/simplePatch')
-	.reply(200, { test: true })
-	.put('/simplePut')
-	.reply(200, { test: true })
-	.post('/simplePost')
-	.reply(200, { test: true })
-	.get('/getQuery')
-	.query({ foo: 'bar', hello: 'world' })
-	.reply(200, { test: true })
-	.get('/getAuth')
-	.times(3)
-	.reply(200, function handler() {
-		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-		return { auth: this.req.headers.authorization?.[0] ?? null };
-	})
-	.get('/getReason')
-	.times(3)
-	.reply(200, function handler() {
-		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-		return { reason: this.req.headers['x-audit-log-reason']?.[0] ?? null };
-	})
-	.post('/urlEncoded')
-	.reply(200, (_, body) => body)
-	.post('/postEcho')
-	.reply(200, (_, body) => body)
-	.post('/postFile')
-	.times(5)
-	.reply(200, (_, body) => ({
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-		body: body
-			.replace(/\r\n/g, '\n')
-			.replace(/-+\d+-*\n?/g, '')
-			.trim(),
-	}))
-	.delete('/channels/339942739275677727/messages/392063687801700356')
-	.reply(200, { test: true })
-	.delete(`/channels/339942739275677727/messages/${newSnowflake}`)
-	.reply(200, { test: true })
-	.get('/request')
-	.times(2)
-	.reply(200, { test: true });
+// @discordjs/rest uses the `content-type` header to detect whether to parse
+// the response as JSON or as an ArrayBuffer.
+const responseOptions: MockInterceptor.MockResponseOptions = {
+	headers: {
+		'content-type': 'application/json',
+	},
+};
+
+/*
+const mockAgent = new MockAgent();
+mockAgent.disableNetConnect();
+setGlobalDispatcher(mockAgent);
+
+const mockPool = mockAgent.get('http://localhost:3000');
+mockPool.intercept({ path: '/foo', method: 'GET' }).reply(200, 'foo');
+
+const res = await fetch('http://localhost:3000/foo');
+
+console.log(res.status, await res.text()); // 200 foo
+*/
```

Comment:
Is this supposed to be left there?

Reply:
It's pretty helpful for people who haven't used undici mocks before IMO. I used that as a basic template for everything.

Reply:
Then you can document it as such, a template.

I won't doubt that any of us will find that comment and delete it thinking it was just some left-overs.

Reply:
I ended up removing it, the tests show how to properly use mocks.